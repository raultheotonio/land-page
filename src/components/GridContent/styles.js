import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58=rem;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhage} 0;
  `}
`;
