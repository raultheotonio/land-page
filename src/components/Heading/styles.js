import styled, { css } from 'styled-components';

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.sizes.xlarge};
  }
`;

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
  `}
`;
