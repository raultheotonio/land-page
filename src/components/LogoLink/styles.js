import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${() => css`
    display: flex;
    align-items: center;
    height: 10rem;

    text-decoration: none;
    color: inherit;
    > img {
      max-height: 3rem;
    }
  `}
`;
