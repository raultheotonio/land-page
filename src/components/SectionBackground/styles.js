import styled, { css } from 'styled-components';

const containerBackgroundActivate = (theme) => css`
  background: ${theme.colors.primarycolor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primarycolor};

    ${background && containerBackgroundActivate(theme)}
    display: flex;
    min-height: 100vh;
    align-items: center;
  `}
`;
