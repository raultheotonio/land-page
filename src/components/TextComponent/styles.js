import styled, { css } from 'styled-components';

export const Container = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.sizes.medium};
  `}
`;
