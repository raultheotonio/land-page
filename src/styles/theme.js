export const theme = {
  colors: {
    primarycolor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
    darkBg: '#0A1128',
    mediumGray: '#DDDDDD',
  },
  fonts: {
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif;",
    },
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlage: '4.8rem',
    huge: '5.6rem',
    xhuge: '5.4rem',
  },
  sizes: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlage: '4.8rem',
    huge: '5.6rem',
    xhuge: '5.4rem',
  },

  media: {
    lteMedium: '(max-width: 768px)',
  },
};
