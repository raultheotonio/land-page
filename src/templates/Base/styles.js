import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 5.4rem;

    @media ${theme.media.letMedium} {
      padding-top: 0rem;
    }
  `}
`;
