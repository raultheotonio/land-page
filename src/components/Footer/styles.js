import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.5rem solid %{theme.colors.mediumGray};
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.sizes.small};
    }

    ${TextComponent} {
      font-size: ${theme.sizes.small};
    }

    ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
