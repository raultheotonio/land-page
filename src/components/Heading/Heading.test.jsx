import { screen } from '@testing-library/react';
import { Heading } from '.';
import { theme } from '../../styles/theme';
import renderTheme from '../../utils/RenderTheme';
import 'jest-styled-components';

describe('<Heading/>', () => {
  it('should render with default values', () => {
    renderTheme(
      <Heading uppercase={false} as="h1" colorDark size="small">
        texto
      </Heading>
    );

    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.sizes.small,
      'text-transform': 'none',
    });
  });

  it('should render with white color', () => {
    renderTheme(
      <Heading uppercase={false} as="h1" colorDark={false} size="small">
        texto
      </Heading>
    );

    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('should render a snapshot', () => {
    const { container } = renderTheme(
      <Heading uppercase={false} as="h1" colorDark={false} size="small">
        texto
      </Heading>
    );

    expect(container).toMatchSnapshot();
  });
});
