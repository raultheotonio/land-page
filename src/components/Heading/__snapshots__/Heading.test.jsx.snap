// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading/> should render a snapshot 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1.6rem;
  text-transform: none;
}

<div>
  <div>
    <h1
      class="c0"
    >
      texto
    </h1>
  </div>
</div>
`;
